internal/celoapi/backend.go

```diff
@@ -7,91 +7,87 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/exchange"
-	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/lru"
 	"github.com/ethereum/go-ethereum/contracts"
-	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/rpc"
 )
 
-type Ethereum interface {
-	BlockChain() *core.BlockChain
+func NewCeloBackend(b ethapi.Backend) *CeloBackend {
+	return &CeloBackend{
+		Backend:            b,
+		exchangeRatesCache: lru.NewCache[common.Hash, common.ExchangeRates](128),
+	}
 }
 
-type CeloAPI struct {
-	ethAPI *ethapi.EthereumAPI
-	eth    Ethereum
-}
+// CeloBackend is a wrapper for the ethapi.Backend, that provides additional Celo specific
+// functionality. CeloBackend is mainly passed to the JSON RPC services and provides
+// an easy way to make extra functionality available in the service internal methods without
+// having to change their call signature significantly.
+// CeloBackend keeps a threadsafe LRU cache of block-hash to exchange rates for that block.
+// Cache invalidation is only a problem when an already existing blocks' hash
+// doesn't change, but the rates change. That shouldn't be possible, since changing the rates
+// requires different transaction hashes / state and thus a different block hash.
+// If the previous rates change during a reorg, the previous block hash should also change
+// and with it the new block's hash.
+// Stale branches cache values will get evicted eventually.
```

Comment:
nitpick: you can be more confident in the statements and say

```suggestion
// doesn't change, but the rates change. That isn't possible, since changing the rates
// requires different transaction hashes / state and thus a different block hash.
// If the previous rates change during a reorg, the previous block hash will also change
// and with it the new block's hash.
// Stale branches cache values will get evicted eventually.
```